src/components/Transactions/TransactionTrade.tsx

```diff
@@ -0,0 +1,178 @@
+import { CheckCircleIcon, WarningTwoIcon } from '@chakra-ui/icons'
+import { Box, Collapse, Flex, Link, SimpleGrid, Tag } from '@chakra-ui/react'
+import { Asset, chainAdapters } from '@shapeshiftoss/types'
+import dayjs from 'dayjs'
+import { useState } from 'react'
+import { FaExchangeAlt } from 'react-icons/fa'
+import { Amount } from 'components/Amount/Amount'
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
+import { IconCircle } from 'components/IconCircle'
+import { MiddleEllipsis } from 'components/MiddleEllipsis/MiddleEllipsis'
+import { Row } from 'components/Row/Row'
+import { RawText, Text } from 'components/Text'
+import { TxDetails } from 'hooks/useTxDetails/useTxDetails'
+import { fromBaseUnit } from 'lib/math'
+
+export const TransactionTrade = ({
+  txDetails,
+  activeAsset
+}: {
+  txDetails: TxDetails
+  activeAsset?: Asset
+}) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const toggleOpen = () => setIsOpen(!isOpen)
+
+  return (
+    <>
+      <Flex
```

Comment:
with these three components, we might be able to keep some of this parenty/top level DOM stuff in a generic TxRow component, and inject the specific children/contents via render props, or whatever the cool kids are doing these days

@reallybeard can help in terms of how much will be common vs specific, as he'll be designing (and possible providing) the specific components we use to render specific txs.

overall thought this is spot on.

Reply:
Resolving as per https://github.com/shapeshift/web/pull/811#issuecomment-1015934141